scipy/interpolate/interpolate.py

```diff
@@ -1712,8 +1712,10 @@ def from_derivatives(cls, xi, yi, orders=None, extrapolate=None):
         # global poly order is k-1, local orders are <=k and can vary
         try:
             k = max(len(yi[i]) + len(yi[i+1]) for i in range(m))
-        except TypeError:
-            raise ValueError("Using a 1-D array for y? Please .reshape(-1, 1).")
+        except TypeError as e:
+            raise ValueError(
+                "Using a 1-D array for y? Please .reshape(-1, 1)."
```

Comment:
similar comment here, I feel like a fair bit of the error messaging in SciPy is written assuming an interactive session is being used. This is definitely an important use case but certainly not the only way SciPy is used. 

To be clear, I am not suggesting ***all*** of the error messages in this PR should be changes to include interpolation but some would clearly aid the end user in root causing source of the raised error.

Reply:
Same answer as the other comment.